ext/SpeedyWeatherCUDAExt/legendre.jl

```diff
@@ -0,0 +1,404 @@
+# convert i, j indices of a matrix (here 0-based l,m though...) to a single 1-based running index
+import SpeedyWeather.LowerTriangularMatrices: ij2k
+
+# range of the running indices lm in a l-column (degrees of spherical harmonics)
+# given the column index m (order of harmonics) 
+get_lm_range(m, lmax) = ij2k(2*m - 1, m, lmax):ij2k(lmax+m, m, lmax)
+get_2lm_range(m, lmax) = 2*ij2k(2*m - 1, m, lmax)-1:2*ij2k(lmax+m, m, lmax)
```

Comment:
Those utility functions are quite useful outside of the GPU legendre transform as well (e.g. in some of the regular unit tests as well). We could also just move them into the regular LowerTriangularMatrices submodule.